src/dispatcher/dispatcher.js

```diff
@@ -46,139 +39,179 @@ let channel;
+   * @param {Object} rabbitConf 'amqplib connection object (see https://amqp-node.github.io/amqplib/channel_api.html#connect)
+   */
+  constructor(workerQueue, resultQueue, rabbitConf) {
+    this.rabbitConf = rabbitConf;
+    this.workerQueue = workerQueue;
+    this.resultQueue = resultQueue;
+
+    this.connection = undefined;
+    this.channel = undefined;
+  }
 
-  channel.assertQueue('DeadLetterQueue', { durable: true });
+  /**
+   * Init the dispatching process.
+   */
+  async init() {
+    this.connection = await amqp.connect(this.rabbitConf);
 
-  channel.bindQueue('DeadLetterQueue', 'DeadLetterExchange', '');
+    this.channel = await this.connection.createChannel();
 
-  channel.assertQueue(workerQueue, {
-    durable: true,
-    arguments: {
-      'x-dead-letter-exchange': 'DeadLetterExchange'
-    }
-  });
-  channel.assertQueue(resultQueue, {
-    durable: true
-  });
+    this.channel.assertExchange('DeadLetterExchange', 'fanout', {
+      durable: true,
+      autoDelete: false
+    });
 
-  log(
-    `Dispatcher waiting for messages in ` + `${workerQueue} and ${resultQueue}.`
-  );
+    this.channel.assertQueue('DeadLetterQueue', { durable: true });
 
-  channel.consume(workerQueue, handleNextTask, {
-    noAck: false
-  });
+    this.channel.bindQueue('DeadLetterQueue', 'DeadLetterExchange', '');
 
-  channel.consume(resultQueue, handleResults, {
-    noAck: false
-  });
-};
+    this.channel.assertQueue(this.workerQueue, {
+      durable: true,
+      arguments: {
+        'x-dead-letter-exchange': 'DeadLetterExchange'
+      }
+    });
+    this.channel.assertQueue(this.resultQueue, {
+      durable: true
+    });
 
-/**
- * Determine the next task from job list and sends it to worker queue.
- * If there is no task left, finish the execution with success.
- * If an error is detected in a result message, we `nack` the job message which
- * then will be forwarded to the dead letter exchange.
- * @param {Object} msg The message
- */
-const handleNextTask = (msg) => {
-  let nextTaskEntry;
+    logger.debug(
+      `Dispatcher waiting for messages in ${this.workerQueue} and ${this.resultQueue}.`
+    );
 
-  try {
-    const job = JSON.parse(msg.content.toString());
-    const chain = job.job;
-    log('Received a job configuration ...');
+    this.channel.consume(this.workerQueue, this.#createHandleNextTaskFun(), {
+      noAck: false
+    });
 
-    if (job.error) {
-      throw job.error;
-    }
+    this.channel.consume(this.resultQueue, this.#createHandleResultFun(), {
+      noAck: false
+    });
+  }
 
-    // validate
-    if (!chain || chain.length < 1) {
-      throw 'Invalid arguments given' + job;
-    }
+  /**
+   * Creates a function that is executed when next task is available.
+   *
+   * Function needs to be created in this way to ensure it has access
+   * to the channel within its scope.
+   *
+   * @returns {Function} The function
+   */
+  #createHandleNextTaskFun() {
```

Comment:
#?

Reply:
that's apparently the way to make functions private https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields